src/main/java/org/assertj/core/internal/Arrays2D.java

```diff
@@ -0,0 +1,155 @@
+  }
+
+  void assertHasSize(AssertionInfo info, Object array, int expectedSize) {
+    assertNotNull(info, array);
+    int sizeOfActual = sizeOf(array);
+    CommonValidations.checkSizes(array, sizeOfActual, expectedSize, info);
+  }
+
+  void assertHasSize(AssertionInfo info, Failures failures, Object array, int expectedSize, int firstDimensionIndex) {
+    assertNotNull(info, array);
+    int sizeOfActual = sizeOf(array);
+    checkSizes(array, failures, sizeOfActual, expectedSize, info, firstDimensionIndex);
+  }
+
+  private void checkSizes(Object actual, Failures failures, int sizeOfActual, int sizeOfOther, AssertionInfo info,
+    int firstDimensionIndex) {
+    if (sizeOfActual != sizeOfOther) throw failures.failure(info, shouldHaveSize(actual, sizeOfActual, sizeOfOther,
+      firstDimensionIndex));
+  }
+
+  public void assertHasSameSizeAs(AssertionInfo info, Object array, Object other) {
+    assertNotNull(info, array);
+    assertIsArray(info, array);
+    assertIsArray(info, other);
+    hasSameSizeAsCheck(info, array, other, sizeOf(array));
+  }
+
+  void assertContains(AssertionInfo info, Failures failures, Object array, Object value, Index index) {
+    assertNotNull(info, array);
+    assertNotEmpty(info, failures, array);
+    checkIndexValueIsValid(index, sizeOf(array) - 1);
+    Object actualElement = Array.get(array, index.value);
+    if (!areEqual(actualElement, value))
+      throw failures.failure(info, shouldContainAtIndex(array, value, index, Array.get(array, index.value),
+                                                        comparisonStrategy));
+  }
+
+  void assertNotEmpty(AssertionInfo info, Failures failures, Object array) {
+    assertNotNull(info, array);
+    if (isArrayEmpty(array)) throw failures.failure(info, shouldNotBeEmpty());
+  }
+
+  void assertNotEmpty(AssertionInfo info, Failures failures, Object array, ErrorMessageFactory failureMessage) {
+    assertNotNull(info, array);
+    if (isArrayEmpty(array)) throw failures.failure(info, failureMessage);
+  }
+
+  void assertDoesNotContain(AssertionInfo info, Failures failures, Object array, Object value, Index index) {
+    assertNotNull(info, array);
+    checkIndexValueIsValid(index, Integer.MAX_VALUE);
+    if (index.value >= sizeOf(array)) return;
+    if (areEqual(Array.get(array, index.value), value))
+      throw failures.failure(info, shouldNotContainAtIndex(array, value, index, comparisonStrategy));
+  }
+
+  private boolean areEqual(Object actual, Object other) {
+    return comparisonStrategy.areEqual(actual, other);
+  }
+
+  private static boolean isArrayEmpty(Object array) {
+    return sizeOf(array) == 0;
+  }
+
+  private static void assertNotNull(AssertionInfo info, Object array) {
+    Objects.instance().assertNotNull(info, array);
+  }
+
+  private static int sizeOf(Object array) {
+    if (array instanceof Object[]) return ((Object[]) array).length;
+    return getLength(array);
+  }
```

Comment:
Refactor: this should be in `org.assertj.core.util.Arrays` and the replace same method in `org.assertj.core.internal.Arrays` with unit test like in `Arrays_array_Test`

Reply:
I'm not sure if I understood you correctly, so I deleted this method and referenced those from `Arrays` class which was identical.

Reply:
fine by me